system-addon/test/functional/mochitest/browser_highlights_section.js

```diff
@@ -21,10 +19,13 @@ async function add_highlights_bookmark(count) {
   for (let placeInfo of bookmarks) {
     await PlacesTestUtils.addVisits(placeInfo.url);
   }
+
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
```

Comment:
Moving this code doesn't make it any easier to understand. Probably better to put it in a documented function that can be reimplemented later to not need the pref changes.

Reply:
The move (triggering pref changes only once) actually fixes the crash. (that's why I moved it)
https://treeherder.mozilla.org/#/jobs?repo=try&revision=c57070ed4152c44f76ad339b3b0306b04c79ad63